config/routes.rb

```diff
@@ -1,13 +1,13 @@
+def valid_admin_host?(host)
```

Comment:
It's not ideal having this defined here as this will become globally available to the whole application. I imagine it'd be betteras a static method of the AdminRequest class so that you can call it with AdminRequest.valid_admin_host? You'd then need to change the matches method to use `self.class.valid_admin_host?()`

Reply:
> Or you could inline it below with:
> 
> ```
> constraints: ->(request) { request.host.starts_with? "whitehall-admin" }
> ```

Checks also happen within the admin namespace routes that use the class method, so just saves some repetition.